tutorial/source/svi_flow_guide.ipynb

```diff
@@ -0,0 +1,236 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# SVI with a Normalizing Flow guide\n",
+    "\n",
+    "Thanks to their expressiveness, normalizing flows (see [normalizing flow introduction](normalizing_flows_intro.ipynb)) are great guide candidates for stochastic variational inference (SVI). This notebook demonstrates how to perform amortized SVI with a normalizing flow as guide."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pyro\n",
+    "import torch\n",
+    "import zuko  # pip install zuko\n",
+    "\n",
+    "from corner import corner, overplot_points  # pip install corner\n",
+    "from pyro.contrib.zuko import Zuko2Pyro\n",
+    "from pyro.optim import ClippedAdam\n",
+    "from pyro.infer import SVI, Trace_ELBO\n",
+    "from torch import Tensor"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Model\n",
+    "\n",
+    "We define a simple non-linear model $p(x | z)$ with a standard Gaussian prior $p(z)$ over the latent variables $z$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "prior = pyro.distributions.Normal(torch.zeros(3), torch.ones(3)).to_event(1)\n",
+    "\n",
+    "def likelihood(z: Tensor):\n",
+    "    mu = z[..., :2]\n",
+    "    rho = z[..., 2].tanh() * 0.99\n",
+    "\n",
+    "    cov = 1e-2 * torch.stack([\n",
+    "        torch.ones_like(rho), rho,\n",
+    "        rho, torch.ones_like(rho),\n",
+    "    ], dim=-1).unflatten(-1, (2, 2))\n",
+    "\n",
+    "    return pyro.distributions.MultivariateNormal(mu, cov)\n",
+    "\n",
+    "def model(x: Tensor):\n",
+    "    with pyro.plate(\"batch\", len(x)):\n",
+    "        z = pyro.sample(\"z\", prior)\n",
+    "\n",
+    "        with pyro.plate(\"obs\", 5):\n",
+    "            pyro.sample(\"x\", likelihood(z), obs=x.transpose(0, 1))"
```

Comment:
are all these transposes necessary?

Reply:
Either there or in the guide.

Reply:
it seems like this could be re-organized more elegantly to reduce the number of transposes

Reply:
I think I can reduce the number of `transpose` from 2 to 1.